diff --git a/dist/components/autocomplete/autocomplete.js b/dist/components/autocomplete/autocomplete.js
index 1fe6552974e6e6235e948bb18e1959ba191cf881..fe91a1df13bdef80da233f59f9e82c9a2526a8bb 100644
--- a/dist/components/autocomplete/autocomplete.js
+++ b/dist/components/autocomplete/autocomplete.js
@@ -4,10 +4,11 @@ import { mergeRefs, useResizeObserver } from '@react-aria/utils';
 import React__default, { useRef, createContext, use, useState, useCallback } from 'react';
 import { Autocomplete } from 'react-aria-components/Autocomplete';
 import { Button } from 'react-aria-components/Button';
-import { Dialog } from 'react-aria-components/Dialog';
+import { Dialog, DialogContext } from 'react-aria-components/Dialog';
 import { Group } from 'react-aria-components/Group';
-import { Popover } from 'react-aria-components/Popover';
+import { Popover, PopoverContext as RACPopoverContext } from 'react-aria-components/Popover';
 import { Select, SelectStateContext, SelectValue } from 'react-aria-components/Select';
+import { useSlottedContext } from 'react-aria-components/slots';
 import { dataAttr } from '../../utils/assertion.js';
 import { composeTwRenderProps, composeSlotClassName } from '../../utils/compose.js';
 import { dom } from '../../utils/dom.js';
@@ -180,6 +181,13 @@ const AutocompletePopover = ({
     slots,
     triggerRef
   } = use(AutocompleteContext);
+  const racPopoverContext = useSlottedContext(RACPopoverContext);
+  const dialogContext = racPopoverContext ? {
+    "aria-labelledby": racPopoverContext["aria-labelledby"]
+  } : {
+    "aria-label": props["aria-label"] ?? (props["aria-labelledby"] ? undefined : "Autocomplete options"),
+    "aria-labelledby": props["aria-labelledby"]
+  };
   const [triggerWidth, setTriggerWidth] = useState(null);
   const onResize = useCallback(() => {
     if (triggerRef.current) {
@@ -204,10 +212,13 @@ const AutocompletePopover = ({
         "--trigger-width": triggerWidth,
         ...(typeof style === "object" && style !== null ? style : {})
       },
-      children: /*#__PURE__*/jsx(Dialog, {
-        className: slots?.popoverDialog(),
-        "data-slot": "autocomplete-popover-dialog",
-        children: children
+      children: /*#__PURE__*/jsx(DialogContext.Provider, {
+        value: dialogContext,
+        children: /*#__PURE__*/jsx(Dialog, {
+          className: slots?.popoverDialog(),
+          "data-slot": "autocomplete-popover-dialog",
+          children: children
+        })
       })
     })
   });
